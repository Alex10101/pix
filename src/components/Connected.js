// import React from 'react';
import { connect } from 'react-redux';
import { doVoidAction } from 'store/actions/void';

/**
 * @return {string}
 */
function Account({ Void, doVoid }) {
  console.log(Void);
  doVoid(true);
  return '';
}

const mapStateToProps = ({ Void }) => ({ Void });
const mapDispatchToProps = { doVoid: doVoidAction };

export default connect(mapStateToProps, mapDispatchToProps)(Account);
