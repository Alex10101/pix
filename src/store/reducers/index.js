import { combineReducers } from 'redux';
import Void from './void';

export default combineReducers({ Void });
