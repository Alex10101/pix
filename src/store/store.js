import { createStore } from 'redux';
import reducers from './reducers';

const store = (initialState) => createStore(reducers, initialState);

export default store;
