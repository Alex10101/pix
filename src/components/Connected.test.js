import React from 'react';
import { create, act } from 'react-test-renderer';

import { Provider } from 'react-redux';
import store from 'store/store';
import Connected from './Connected';

describe('Void', () => {
  let snapshot;

  const testingObj = {
    Void: 123
  };

  const testingStore = store(testingObj);

  it("' s redux should handle preloaded state", () => {
    // console.log(JSON.stringify(testingStore.getState()));
    // console.log(JSON.stringify(testingObj));
    expect(JSON.stringify(testingStore.getState())).toBe(JSON.stringify(testingObj));
  });

  it('should change redux state', async () => {
    await act(async () => {
      snapshot = create(
        <Provider store={testingStore}>
          <Connected />
        </Provider>
      );
    });

    expect(snapshot.toJSON()).toMatchSnapshot();
    expect(testingStore.getState().Void).toBe(true);
  });
});
