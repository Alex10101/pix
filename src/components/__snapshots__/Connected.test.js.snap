// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Void should change redux state 1`] = `""`;
